### PR TITLE
Update to JanePhp v5

### PR DESCRIPTION
**Pull request by pyrech**
No BC break \o/

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,13 +5,13 @@ cs_dry_run: ## Display CS violations without fixing it
...
 spec_diff: ## detect diff between spec and SDK
-	CI_TEST=1 vendor/bin/jane-openapi generate --config-file=.jane-openapi.php && diff generated_ci generated
+	CI_TEST=1 vendor/bin/jane-openapi generate --config-file=.jane-openapi.php && vendor/bin/php-cs-fixer fix && diff generated_ci generated
 
 test: ## Run test suite
 	./vendor/bin/simple-phpunit
 
 jane: ## Generate the SDK
-	rm -rf generated/* && vendor/bin/jane-openapi generate --config-file=.jane-openapi.php
+	rm -rf generated/* && vendor/bin/jane-openapi generate --config-file=.jane-openapi.php && vendor/bin/php-cs-fixer fix
```
Comment (Korbeil) [inline Makefile]: The deletion of source file is already done by `clean-generated` command option (which is `true` by default)

Comment (Korbeil) [inline Makefile]: You can configure Jane to fix your code with `use-fixer` to `true` (default is `false`)
Then you can give him a custom cs-fixer configuration with `fixer-config-file` and pointing to your configuration file.

File: composer.json [hunk 2]
```diff
@@ -22,15 +22,15 @@
...
+        "php": ">= 7.2",
+        "jane-php/open-api-runtime": "^5.3",
         "psr/http-client-implementation": "*",
         "php-http/client-common": "^1.9 || ^2.0",
         "php-http/discovery": "^1.7",
         "php-http/multipart-stream-builder": "^1.1"
     },
     "require-dev": {
-        "jane-php/jane-php": "^4.5.3",
+        "jane-php/open-api-2": "^5.3",
```
Comment (Korbeil) [inline composer.json]: You should restrict to `^5.3.1`, `v5.3.0` was a bit buggy and should be avoided

File: doc/updating-sdk.md [hunk 3]
```diff
@@ -50,4 +50,5 @@ SDK:
 
 ```bash
 vendor/bin/jane-openapi generate --config-file=.jane-openapi.php
+vendor/bin/php-cs-fixer fix
```
Comment (Korbeil) [inline doc/updating-sdk.md]: Not needed if you do cs-fix during Jane generation (look at https://github.com/jolicode/slack-php-api/pull/73/files#r382296204)

File: generated/Client.php [hunk 4]
```diff
@@ -1,11 +1,12 @@
 <?php
 
-declare(strict_types=1);
```
Comment (Korbeil) [inline generated/Client.php]: Since your cs-fixer configuration does not enforce it, you will loose all `strict_types`, that's kind of sad here :/

**Comment by pyrech** [comment]
PR ready :slightly_smiling_face:

File: .jane-openapi.php [hunk 5]
```diff
@@ -13,4 +13,7 @@
     'reference' => true,
     'strict' => false,
     'client' => 'psr18',
+    'clean-generated' => true,
```
Comment (Korbeil) [inline .jane-openapi.php]: default value is `true`, this option is not usefull here

Comment (pyrech) [reply .jane-openapi.php]: I prefer explicitness here :wink:

**Pull request merged**